plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergAnalyze.java

```diff
@@ -0,0 +1,570 @@
+                        "('regionkey', null, 15, 0, null, '0', '14')," +
+                        "('comment', null, 50, 0, null, null, null)," + // result of previous analyze
+                        "('name', null, 50, 0, null, null, null)," + // result of previous analyze
+                        "(null, null, null, null, 75, null, null)");
+
+        // analyze all columns
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 75, 0, null, '0', '74')," +
+                        "('regionkey', null, 15, 0, null, '0', '14')," +
+                        "('comment', null, 75, 0, null, null, null)," +
+                        "('name', null, 75, 0, null, null, null)," +
+                        "(null, null, null, null, 75, null, null)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDropExtendedStats()
+    {
+        String tableName = "test_drop_extended_stats";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+
+        String baseStats = "VALUES"
+                + "('nationkey', null, null, 0, null, '0', '24'),"
+                + "('regionkey', null, null, 0, null, '0', '4'),"
+                + "('comment', null, null, 0, null, null, null),"
+                + "('name',  null, null, 0, null, null, null),"
+                + "(null,  null, null, null, 25, null, null)";
+        String extendedStats = "VALUES"
+                + "('nationkey', null, 25, 0, null, '0', '24'),"
+                + "('regionkey', null, 5, 0, null, '0', '4'),"
+                + "('comment', null, 25, 0, null, null, null),"
+                + "('name',  null, 25, 0, null, null, null),"
+                + "(null,  null, null, null, 25, null, null)";
+
+        assertQuery("SHOW STATS FOR " + tableName, baseStats);
+
+        // Update stats to include distinct count
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery("SHOW STATS FOR " + tableName, extendedStats);
+
+        // Dropping extended stats clears distinct count and leaves other stats alone
+        assertUpdate("ALTER TABLE " + tableName + " EXECUTE DROP_EXTENDED_STATS");
+        assertQuery("SHOW STATS FOR " + tableName, baseStats);
+
+        // Re-analyzing should work
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery("SHOW STATS FOR " + tableName, extendedStats);
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testDropMissingStats()
+    {
+        String tableName = "test_drop_missing_stats";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+
+        // When there are no extended stats, the procedure should have no effect
+        assertUpdate("ALTER TABLE " + tableName + " EXECUTE DROP_EXTENDED_STATS");
```

Comment:
to show this shouldn't you run `show stats` before executing drop ?  Now we only see results after analyze which doesn't prove the point imho

Reply:
this is covered by other tests, so no need to re-test this here